app/api/retrieval/keys/route.ts

```diff
@@ -0,0 +1,53 @@
+export type KeyTypeT =
+  | "OPENAI_API_KEY"
+  | "ANTHROPIC_API_KEY"
+  | "GOOGLE_GEMINI_API_KEY"
+  | "MISTRAL_API_KEY"
+  | "PERPLEXITY_API_KEY"
+  | "AZURE_OPENAI_API_KEY"
+
+// returns true if the key is found in the environment variables
+function isUsingEnvironmentKey(type: KeyTypeT) {
+  return Boolean(process.env[type])
+}
+export enum VALID_KEYS {
+  OPENAI_API_KEY = "OPENAI_API_KEY",
+  ANTHROPIC_API_KEY = "ANTHROPIC_API_KEY",
+  GOOGLE_GEMINI_API_KEY = "GOOGLE_GEMINI_API_KEY",
+  MISTRAL_API_KEY = "MISTRAL_API_KEY",
+  PERPLEXITY_API_KEY = "PERPLEXITY_API_KEY",
+  AZURE_OPENAI_API_KEY = "AZURE_OPENAI_API_KEY"
+}
+
+export async function POST(request: Request) {
```

Comment:
I believe the logic can be enhanced further by incorporating reusable functions.

e.g: 
```
function createResponse(data, status): Response {
  return new Response(JSON.stringify(data), {
    status,
    headers: {
      "Content-Type": "application/json"
    }
  });
}
```

Reply:
Usage 

```
 if (!key) {
      return createResponse({ error: "Key type is required" }, 400);
    }
```

Reply:
done in b4ea508